src/tiledb/cloud/sql/tiledb_cursor.py

```diff
@@ -0,0 +1,116 @@
+import pyarrow as pa
+
+from tiledb.cloud.sql._execution import exec
+from tiledb.cloud.sql.db_api_exceptions import DataError
+from tiledb.cloud.sql.db_api_exceptions import NotSupportedError
+from tiledb.cloud.sql.db_api_exceptions import ProgrammingError
+
+
+def _get_db_type(dtype):
+    if pa.types.is_timestamp(dtype):
+        return "DATETIME"
+    elif pa.types.is_boolean(dtype):
+        return "BOOLEAN"
+    elif pa.types.is_string(dtype):
+        return "STRING"
+    elif pa.types.is_integer(dtype) or pa.types.is_floating(dtype):
+        return "NUMBER"
+    return "UNKNOWN"
+
+
+class Cursor:
+    def __init__(self):
+        self._results = None
+        self._row_index = 0
+        self.arraysize = 1
+        self.rowcount = -1
+        self._description_cache = None
+
+    def executemany(self, query, seq_of_parameters):
+        for params in seq_of_parameters:
+            self.execute(query, params)
+
+    def execute(self, query, params=()):
+        try:
+            self._description_cache = None
+            self._results = exec(query=query, raw_results=True)
+            self.rowcount = len(self._results)
+        except Exception as e:
+            self._results = None
+            raise DataError(f"Error executing query: {e}") from e
+
+    def fetchmany(self, size=-1):
+        if not self._results:
+            return None
+
+        if size == -1:
+            size = self.arraysize
+
+        if size + self._row_index > len(self._results):
+            # give all that is remaining
+            size = len(self._results) - self._row_index
+
+        if size == 0:
+            # There are no more records
+            return []
+
+        rows = self._results[self._row_index : self._row_index + size]
+        self._row_index += size
+        return rows.to_pylist()
+
+    def setinputsizes(self, sizes):
+        raise NotSupportedError("Operation not supported")
+
+    def setoutputsize(self, size, column):
+        raise NotSupportedError("Operation not supported")
+
+    def fetchone(self):
+        result = self.fetchmany(1)
+        if len(result) == 0:
+            return None
+        return result[0]
+
+    def description(self):
+        if self._description_cache is not None:
+            return self._description_cache
+
+        if not self._results:
+            return None
+
+        description = [
+            (field.name, _get_db_type(field.type)) for field in self._results.schema
```

Comment:
This needs to be a 7-tuple; the rest of the entries will all be None:

```python
(field.name, _get_db_type(field.type), None, None, None, None, None)
```

(yeah this looks a bit ridiculous)